local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -191,67 +179,77 @@ bool WaypointGenerator::withinGoalRadius() {
 
 // when taking off, first publish waypoints to reach the goal altitude
 void WaypointGenerator::reachGoalAltitudeFirst() {
-  output_.goto_position = planner_info_.offboard_pose.pose.position;
-  output_.goto_position.z = pose_.pose.position.z + 0.5;
-
-  // if goal lies directly overhead, do not yaw
-  float goal_acceptance_radius = 1.0f;
-  if ((goal_ - toEigen(pose_.pose.position)).norm() < goal_acceptance_radius &&
-      std::isfinite(curr_yaw_)) {
-    new_yaw_ = curr_yaw_;
-  }
-
-  // constrain speed
-  Eigen::Vector3f pose_to_wp =
-      toEigen(output_.goto_position) - toEigen(pose_.pose.position);
-
-  if (pose_to_wp.norm() > 0.01f) pose_to_wp.normalize();
+  // goto_position is a unit vector pointing straight up from current location
+  output_.goto_position = pose_.pose.position;
+  output_.goto_position.z += 1.0f;
 
-  pose_to_wp *= planner_info_.min_speed;
-
-  output_.goto_position = toPoint(toEigen(pose_.pose.position) + pose_to_wp);
+  // Will be overwritten by adaptSpeed()
   output_.adapted_goto_position = output_.goto_position;
+
+  // Will be overwritten by smoothWaypoint()
   output_.smoothed_goto_position = output_.goto_position;
-  smoothed_goto_location_ = toEigen(output_.smoothed_goto_position);
 }
 
 void WaypointGenerator::smoothWaypoint(float dt) {
-  if (smoothing_speed_ > 0.01f) {
-    const float P_constant = smoothing_speed_;
-    const float D_constant = 2.f * std::sqrt(P_constant);  // critically damped
-    const Eigen::Vector3f desired_location =
-        toEigen(output_.adapted_goto_position);
-    const Eigen::Vector3f desired_velocity = toEigen(curr_vel_.twist.linear);
-
-    Eigen::Vector3f location_diff = desired_location - smoothed_goto_location_;
-    if (!location_diff.allFinite()) {
-      location_diff = Eigen::Vector3f::Zero();
-    }
-
-    Eigen::Vector3f velocity_diff =
-        desired_velocity - smoothed_goto_location_velocity_;
-    if (!velocity_diff.allFinite()) {
-      velocity_diff = Eigen::Vector3f::Zero();
-    }
+  // In first iteration (very small or invalid dt), initialize the smoothing
+  // to start at the current drone location
+  if (dt <= 0.01) {
+    smoothed_goto_location_ = toEigen(pose_.pose.position);
+    output_.smoothed_goto_position =
+        pose_.pose.position;  // needs to be local position!
+    return;
+  }
 
-    const Eigen::Vector3f p = location_diff * P_constant;
-    const Eigen::Vector3f d = velocity_diff * D_constant;
+  // If the smoothing speed is set to zero, dont smooth, aka use adapted
+  // waypoint
+  // directly
```

Comment:
could you make it in one line manually please

Reply:
done